### PR TITLE
(PC-14472) fix(booking): refactor naming collect by withdrawal

### PR DESCRIPTION
**Pull request by cedricls1**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-14472

## Checklist

I have:

- [ ] Made sure the title of my PR follows the [convention](1) `($jira) $type($scope): $summary`.
- [ ] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** or **github dev name** for any added TODO / FIXME.
- [ ] Added new reusable components to **AppComponents** page and communicate to the team on slack.
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion](2)
- [ ] Made sure translations file is up to date (`yarn translations:extract`)

## Screenshots

| iOS | Android | Mobile - Chrome | Desktop - Chrome | Desktop - Safari |
| --: | ------: | --------------: | ---------------: | ---------------: |
|     |         |                 |                  |                  |

[1]: https://github.com/pass-culture/pass-culture-app-native/blob/master/doc/standards/pr-title.md
[2]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

**Review by bruno-ebstein-pass-culture — APPROVED** [review]

File: src/features/bookings/components/BookingDetailsTicketContent.tsx [hunk 1]
```diff
@@ -152,22 +152,25 @@ const NoTicket = () => {
...
 
-  if (collectType === WithdrawalTypeEnum.no_ticket) {
+  if (withdrawalType === WithdrawalTypeEnum.no_ticket) {
     return <NoTicket />
   }
 
-  if (collectType === WithdrawalTypeEnum.on_site || collectType === WithdrawalTypeEnum.by_email) {
+  if (
+    withdrawalType === WithdrawalTypeEnum.on_site ||
+    withdrawalType === WithdrawalTypeEnum.by_email
```
[the remark this answers is not in the document]
Comment (cedricls1) [reply src/features/bookings/components/BookingDetailsTicketContent.tsx:167]: Les 2 me vont par contre withdrawalType peut être undefined donc ça donnerait ça :

`  if (
    withdrawalType &&
    [WithdrawalTypeEnum.on_site, WithdrawalTypeEnum.by_email].includes(withdrawalType)
  )`

Est-ce que tu préfères ?

[the remark this answers is not in the document]
Comment (bruno-ebstein-pass-culture) [reply src/features/bookings/components/BookingDetailsTicketContent.tsx:167]: s'il est `undefined`, ça sera équivalent à

```ts
[WithdrawalTypeEnum.on_site, WithdrawalTypeEnum.by_email].includes(undefined)
```

donc `undefined` ne sera pas contenu dans le tableau

je ne pense pas qu'il y ait besoin de complexifier la condition

[the remark this answers is not in the document]
Comment (cedricls1) [reply src/features/bookings/components/BookingDetailsTicketContent.tsx:167]: Si je ne l'ajoute pas on a une erreur
Capture d’écran 2022-04-14 à 10 20 52

[the remark this answers is not in the document]
Comment (bruno-ebstein-pass-culture) [reply src/features/bookings/components/BookingDetailsTicketContent.tsx:167]: pour une fois que TS est trop intelligent

**Pull request merged**